core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -700,7 +700,7 @@ public void processPendingDeletes(Index index, IndexSettings indexSettings, Time
                             sleepTime = Math.min(maxSleepTimeMs, sleepTime * 2); // increase the sleep time gradually
                             logger.debug("{} schedule pending delete retry after {} ms", index, sleepTime);
                         } catch (InterruptedException e) {
-                            Thread.interrupted();
+                            Thread.currentThread().interrupt();
```

Comment:
OK. I'm good with  fixing this inconsistency by throwing an InterruptedException (and catching it up  just like we do with LockObtainFailedException